datastore.go

```diff
@@ -337,12 +313,12 @@ func (d *Datastore) Close() error {
 	d.closeOnce.Do(func() {
 		close(d.closing)
 	})
-	d.closeLk.Lock()
-	defer d.closeLk.Unlock()
-	if d.closed {
+	if d.closed.IsSet() {
+		return ErrClosed
+	}
+	if !d.closed.SetToIf(false, true) {
```

Comment:
The first if is not needed, the second if checks closed starts of false

Reply:
good call